nautobot/dcim/migrations/0030_migrate_region_and_site_data_to_locations.py

```diff
@@ -191,39 +228,102 @@ def migrate_site_and_region_data_to_locations(apps, schema_editor):
             cc.locations.add(*region_locs)
             cc.save()
 
+        CustomField = apps.get_model("extras", "customfield")
+        custom_fields = CustomField.objects.filter(content_types__in=[region_ct])
+        for cf in custom_fields:
+            cf.content_types.add(location_ct)
+            cf.save()
+
+        region_locs = Location.objects.filter(location_type=region_lt).exclude(name="Global Region")
+        for location in region_locs:
+            region = Region.objects.get(name=location.name)
+            location._custom_field_data = region._custom_field_data
+        Location.objects.bulk_update(region_locs, ["_custom_field_data"], 1000)
+
+        # Custom Links do not have Region ContentType as one of its ContentType options
+
+        ExportTemplate = apps.get_model("extras", "exporttemplate")
+        export_templates = ExportTemplate.objects.filter(content_type=region_ct)
+        for et in export_templates:
+            et.content_type = location_ct
+        ExportTemplate.objects.bulk_update(export_templates, ["content_type"], 1000)
+
+        ImageAttachment = apps.get_model("extras", "imageattachment")
+        image_attachments = ImageAttachment.objects.filter(content_type=region_ct)
+        for ia in image_attachments:
+            ia.content_type = location_ct
+        ImageAttachment.objects.bulk_update(image_attachments, ["content_type"], 1000)
+
+        JobHook = apps.get_model("extras", "jobhook")
+        job_hooks = JobHook.objects.filter(content_types__in=[region_ct])
+        for jh in job_hooks:
+            jh.content_types.add(location_ct)
+            jh.save()
+
+        Relationship = apps.get_model("extras", "relationship")
+        RelationshipAssociation = apps.get_model("extras", "relationshipassociation")
+
+        src_relationship_associations = RelationshipAssociation.objects.filter(relationship__source_type=region_ct)
```

Comment:
Actually, could this just filter on `source_type`? I forgot that we store the content-type on both the Relationship and the RelationshipAssociation.

Reply:
It can, but with the `update()` statement below, I will end up with an empty queryset in `src_relationship_associations`. It is a weird behavior with Django https://stackoverflow.com/questions/40061794/django-queryset-disappears-after-update-command, that you cannot update the attribute with which you are also filtering the queryset.